BLSQ/iaso
# IA-3456 improve get or create for orgunit (duplicate on uuid)

To minize duplicate and be "ready" when we will introduce the unique index

Related JIRA tickets : IA-3456

## Self proofreading checklist


## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes

see https://github.com/BLSQ/iaso/pull/1652

## How to test

see https://github.com/BLSQ/iaso/pull/1652
but the mobile unit test does that

for a more "real life" reproduction : https://gist.github.com/mestachs/2cc50bf310996d14cc48623a05d42caf

note that we have historical data so we will need an intermediate cleanup
create the a temporary index as in the _notes.md
then really cleanup "old data" and all environments (harder)


## Print screen / video

Upload here print screens or videos showing the changes

## Notes

Things that the reviewers should know: known bugs that are out of the scope of the PR, other trade-offs that were made.
If the PR depends on a PR in bluesquare-components, or merges into another PR (i.o. main), it should also be mentioned here
